### ISSUE TITLE
FullCBFWriter shuffles multiple panels

### ISSUE THREAD

**Issue by biochem-fan** [cctbx/dxtbx#745]
I realized `dxtbx.image_average` shuffles panels in a segmented detector (https://github.com/dials/dials/issues/2712).
My investigation revealed that the problem is in `FullCBFWriter`.

In our detector, panels are numbered as follows.

```
0   1   2   3   4   5
6   7   8   9   10  11
12  x   13  14  15  16
...
```
(The panel 'x' is missing due to a sensor failure)
<img width="1130" alt="image" src="https://github.com/user-attachments/assets/3c9f3c75-8db3-4942-a9cc-1dc5688b44ff">

After converting this NXmx image to a full CBF by `FullCBFWriter`, the panels were shuffled as follows.

```
0   1   2   8   9   10
3   4   5   11  12  13
6   x   7   14  15  16
...
```
<img width="1124" alt="image" src="https://github.com/user-attachments/assets/e834bf16-08d4-4332-95ea-912401c273a0">

In the input NXmx, a 3 x 3 block of panels {0, 1, 2, 6, 7, 8, 12, 13} belongs to one group, {3, 4, 5, 9, 10, 11, 14, 15, 16} to another, and so on. In the output CBF, the axes are wrongly sorted according to this order.

If I remove this grouping (intermediate hierarchy) from the input NXmx file, the CBF output is correct.

Please see the attachment [full-cbf-bug.tar.gz](https://github.com/user-attachments/files/16477970/full-cbf-bug.tar.gz), which contains:

- numbered_panels.h5
- numbered_panels.cbf (wrong)
- numbered_panels-without_hierarchy.h5
- numbered_panels-without_hierarchy.cbf (correct)

The conversion to full CBF was done as follows:

```
from dxtbx.format.cbf_writer import FullCBFWriter
writer = FullCBFWriter("numbered_panels-without_hierarchy.h5")
writer.write_cbf("numbered_panels-without_hierarchy.cbf", index=0)
```

**Comment by biochem-fan** [comment]
I found the cause.

When writing `array_intensities`, [src/dxtbx/format/cbf_writer.py](https://github.com/cctbx/dxtbx/blob/main/src/dxtbx/format/cbf_writer.py) assumes that `panelkeys` and `panelnames` constructed by [recursive_setup_basis_dict()](https://github.com/cctbx/dxtbx/blob/3102224d05d33616855936453af82f6849c5b93b/src/dxtbx/format/cbf_writer.py#L219) are in the same order as what `get_raw_data()` returns, but this is not correct in general.

Because @keitaroyam asked: original pixel data have been replaced with rendered texts of panel numbers for ease of investigation by the following code.
```
# c.f. https://stackoverflow.com/questions/45947608/rendering-a-unicode-ascii-character-to-a-numpy-array
import numpy as np
from PIL import Image, ImageDraw, ImageFont

font = ImageFont.truetype("DejaVuSans.ttf", size=100, encoding="unic")

HEIGHT = 728
for i in range(71):
    canvas = Image.new("1", [728, 384])
    ImageDraw.Draw(canvas).text((10, 10), str(i), font=font, fill="#FFFFFF")
    f["/entry/data/data"][:, (i * HEIGHT):((i + 1) * HEIGHT), :] = np.asarray(canvas).transpose().astype(np.int32) * 10
```

**Comment by biochem-fan** [comment]
The order is inconsistent with `get_detector`, too.

**Comment by biochem-fan** [comment]
Fixed in [f2838d9](https://github.com/cctbx/dxtbx/commit/f2838d967786ba9e3dda9be64094d839721a47c3).
